pkg/notes/notes_map.go

```diff
@@ -0,0 +1,174 @@
+package notes
+
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v2"
+)
+
+// MapProvider interface that obtains release notes maps from a source
+type MapProvider interface {
+	GetMapsForPR(int) (map[int]*ReleaseNotesMap, error)
+}
+
+// NewProviderFromInitString creates a new map provider from an initialization string
+func NewProviderFromInitString(initString string) (MapProvider, error) {
+	// If init string starts with gs:// return a CloudStorageProvider
+	if initString[0:5] == "gs://" {
```

Comment:
Let's make `gcsPrefix` prefix in the `gcs` package public and re-use it here.

Reply:
I'll send a separate PR with the constants. I'd rather deal with a rebase on just that if the other packages have conflicts.

Reply:
Sounds good